app/serializers/api/declarations_csv_serializer.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require "csv"
+
+module API
+  class DeclarationsCsvSerializer
+    attr_reader :declarations, :view
+
+    ATTRIBUTES_TO_EXCLUDE = %w[type].freeze
+
+    def initialize(declarations, view:)
+      @declarations = declarations
+      @view = view
+    end
+
+    def serialize
+      return if declarations.empty?
+
+      CSV.generate do |csv|
+        csv << headers
+
+        declarations_json.each do |declaration_json|
+          csv << to_row(declaration_json, headers:)
+        end
+      end
+    end
+
+  private
+
+    def declarations_json
+      @declarations_json ||= JSON.parse(DeclarationSerializer.render(declarations, view: :v1)).map(&method(:flatten_hash))
```

Comment:
could we pass in the version here as v1 has a few extra fields to v2?

Reply:
whoops, yes, that was the intention its in this class as `view` but I haven't forwarded it to the serializer; good catch I'll add a test as well

Reply:
[Fixed](https://github.com/DFE-Digital/npq-registration/pull/1450/commits/5c56525edd42d46467903bee90b8dbe0d895ef25)